at_end2end_test/test/e2e_test_utils.dart

```diff
@@ -229,3 +227,15 @@ void _loadTheYaml() {
   ///       port: 1234
   ///     ... etc
 }
+
+extension Utils on SimpleOutboundSocketHandler {
+  Future<int> getVersion() async {
+    await writeCommand('info\n');
+    var version = await read();
+    version = version.replaceAll('data:', '');
+    // Since secondary version has gha<number> appended, remove the gha number from version
+    // Hence using split.
+    var versionStr = jsonDecode(version)['version'].split('+')[0];
+    return int.parse(versionStr.replaceAll('.', ''));
```

Comment:
This will break when we go, let's say, from version 3.0.21 (3021) to version 3.1.1 (311)
Semantically, 3.1.1 is greater than 3.0.21

I'd suggest using https://pub.dev/packages/version

Reply:
This is implemented.